cmd/genesis-gen/main.go

```diff
@@ -0,0 +1,268 @@
+		Name:  "bridge-owner",
+		Usage: "The system bridge contract owner address",
+	}
+	bridgeSignersFlag = cli.StringSliceFlag{
+		Name:  "bridge-signer",
+		Usage: "The system bridge contract signer address. This flag can be used multiple times",
+	}
+	vaultOwnerFlag = cli.StringFlag{
+		Name:  "vault-owner",
+		Usage: "The system vault contract owner address",
+	}
+	outputFileFlag = cli.StringFlag{
+		Name:  "output",
+		Usage: "The output file to write the genesis.json to",
+		Value: "genesis.json",
+	}
+)
+
+func init() {
+	app = flags.NewApp(gitCommit, gitDate, "Generate genesis block for DBSC")
+	app.Flags = []cli.Flag{
+		chainIDFlag,
+		epochSizeFlag,
+		gasLimitFlag,
+		gasUsedFlag,
+		validatorFlags,
+		premineFlags,
+		validatorsetOwnerFlag,
+		bridgeOwnerFlag,
+		bridgeSignersFlag,
+		vaultOwnerFlag,
+		outputFileFlag,
+	}
+	app.Action = utils.MigrateFlags(genesisGen)
+	cli.CommandHelpTemplate = flags.OriginCommandHelpTemplate
+}
+
+func genesisGen(c *cli.Context) error {
+	var (
+		validators    = c.GlobalStringSlice(validatorFlags.Name)
+		premineList   = c.GlobalStringSlice(premineFlags.Name)
+		bridgeSigners = c.GlobalStringSlice(bridgeSignersFlag.Name)
+	)
+
+	if len(validators) == 0 {
+		return fmt.Errorf("no validators specified")
+	}
+	if len(premineList) == 0 {
+		log.Warn("no premine specified")
+	}
+	if !c.GlobalIsSet(validatorsetOwnerFlag.Name) {
+		return fmt.Errorf("system ValidatorSet contract owner address is not specified")
+	}
+
+	bridgeOwner := common.Address{}
+	if c.GlobalIsSet(bridgeOwnerFlag.Name) {
+		bridgeOwner = common.HexToAddress(c.GlobalString(bridgeOwnerFlag.Name))
+	} else {
+		log.Warn("system bridge contract owner address is not specified")
+	}
+
+	vaultOwner := common.Address{}
+	if c.GlobalIsSet(vaultOwnerFlag.Name) {
+		vaultOwner = common.HexToAddress(c.GlobalString(vaultOwnerFlag.Name))
+	} else {
+		log.Warn("system vault contract owner address is not specified")
+	}
+
+	if len(bridgeSigners) == 0 {
+		log.Warn("no bridge signers specified")
+	}
+
+	chainConfig := &params.ChainConfig{
+		ChainID:             new(big.Int).SetUint64(c.GlobalUint64(chainIDFlag.Name)),
+		HomesteadBlock:      big.NewInt(0),
+		EIP150Block:         big.NewInt(0),
+		EIP155Block:         big.NewInt(0),
+		EIP158Block:         big.NewInt(0),
+		IBFTBlock:           nil, // this is dogechain v1 network block, new network should not use this
+		ByzantiumBlock:      big.NewInt(0),
+		ConstantinopleBlock: big.NewInt(0),
+		PetersburgBlock:     big.NewInt(0),
+		IstanbulBlock:       big.NewInt(0),
+		PreportlandBlock:    big.NewInt(0),
+		PortlandBlock:       big.NewInt(0),
+		DetroitBlock:        big.NewInt(0),
+		// TODO: Add hawaii hard fork
+		// HawaiiBlock:      big.NewInt(0),
+		IBFT: &params.IBFTConfig{
+			EpochSize: c.GlobalUint64(epochSizeFlag.Name),
+			Type:      params.IBFTPoS,
+		},
+	}
+
+	genesis := &core.Genesis{
+		Config:     chainConfig,
+		Nonce:      0,
+		Timestamp:  uint64(time.Now().Unix()),
```

Comment:
The timestamp should be configurable? Or, `0` if you want it Mainnet or Devnet compatible.

Reply:
[39cfd61](https://github.com/dogechain-lab/dbsc/pull/17/commits/39cfd616ccee8b5e388dfd2f3678569f9e48415c) fix